benchmarks/physics/rigid_body_3d.gd

```diff
@@ -0,0 +1,95 @@
+extends Benchmark
+
+var box_shape := BoxShape3D.new()
+var sphere_shape := BoxShape3D.new()
+
+func _init() -> void:
+	test_physics = true
+	test_idle = true
+
+
+func setup_scene(scene_root: Node3D) -> void:
+	var camera := Camera3D.new()
+	camera.position.y = 0.3
+	camera.position.z = 1.0
+	camera.rotate_x(-0.8)
+	scene_root.add_child(camera)
+
+func create_box(unique_shape: bool) -> RigidBody3D:
+	var rigid_body := RigidBody3D.new()
+
+	var collision_shape := CollisionShape3D.new()
+	if unique_shape:
+		collision_shape.shape = BoxShape3D.new()
+	else:
+		# Reuse existing shape.
+		collision_shape.shape = box_shape
+	rigid_body.add_child(collision_shape)
+
+	return rigid_body
+
+
+func create_sphere(unique_shape: bool) -> RigidBody3D:
+	var rigid_body := RigidBody3D.new()
+
+	var collision_shape := CollisionShape3D.new()
+	if unique_shape:
+		collision_shape.shape = SphereShape3D.new()
+	else:
+		# Reuse existing shape.
+		collision_shape.shape = sphere_shape
+	rigid_body.add_child(collision_shape)
+
+	return rigid_body
+
+
+func benchmark_7500_rigid_body_3d_shared_box_shape() -> Node3D:
+	var scene_root := Node3D.new()
+	setup_scene(scene_root)
+
+	for i in 7500:
+		var box := create_box(false)
+		box.position.x = randf_range(-50, 50)
+		box.position.z = randf_range(-50, 50)
+		scene_root.add_child(box)
+
+	return scene_root
```

Comment:
You can avoid the code duplication in these four benchmarks by moving it into `setup_scene()`

```suggestion
func benchmark_7500_rigid_body_3d_shared_box_shape() -> Node3D:
	return setup_scene(create_box, false, 7500)
```

First-class functions are a great feature :slightly_smiling_face: